src/components/ec/cuda/kernel/ec_cuda_reduce_ops.h

```diff
@@ -84,7 +84,7 @@ cuFloatComplex operator* (const cuFloatComplex & first,
                 }                                                              \
                 _Pragma("unroll") for (i = 0; i < unroll; i++)                 \
                 {                                                              \
-                    if constexpr (strided) {                                   \
+                    if (strided) {                                             \
```

Comment:
can we use function template specialiation to get same behaviour as if constexp?
```
template<bool strided>
void * get_tmp2(s, s2) {return nullptr}

template<>
void * get_tmp2(....)<true> {return s2[line + warp_size * i + j * ld]}

template<>
void * get_tmp2(....)<false> {return s[1 + j][line + warp_size * i]]}

```